src/sentry/incidents/logic.py

```diff
@@ -117,6 +118,20 @@ def update_incident_status(incident, status, user=None, comment=None):
         return incident
 
 
+def set_incident_seen(incident, user=None):
+    """
+    Updates the status of an Incident and write an IncidentActivity row to log
+    the change. When the status is CLOSED we also set the date closed to the
+    current time and (todo) take a snapshot of the current incident state.
+    """
+    incident_seen, created = IncidentSeen.objects.create_or_update(
+        incident=incident,
+        user=user,
+        last_seen=timezone.now(),
```

Comment:
You'll want to pass `last_seen` as part of the `values` dict here. This allows us to attempt to do an update based on just the `incident` and `user` with just those specific values.